src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +446,82 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool allocation_at_init = !LOG_ISRESTARTED ();
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_total_mem_usage += size;
+
+    m_module[module_idx]->add_stat (size, comp_idx, subcomp_idx, allocation_at_init);
+
+    if (thread_p)
+      {
+	tdes = LOG_FIND_CURRENT_TDES (thread_p);
+	tdes->cur_mem_usage += size;
+      }
   }
 
-  int memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
+  void memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
```

Comment:
I think `add_stat` and `sub_stat` can be integrated into one function. It has the exact same logic, but the sign.
```
add_stat (thread_p, -, -, 100); // add stat
add_stat (thread_p, -, -, -100); // sub stat
```
If you'd like keep the meaning, we can still do share the code like
```
void memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
{
  add_stat (thread_p, stat_id, -size);.
}
```

Then, we loose 1 bit to represent the size, but I don't think it's necessary.
It can be applied to all `add/sub_*` functions.

Reply:
yes, with the change.

Reply:
Done.